src/commands/context/selectDockerContext.ts

```diff
@@ -0,0 +1,31 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ProgressLocation, window } from 'vscode';
+import { ext } from '../../extensionVariables';
+import { localize } from '../../localize';
+import { dockerContextManager, IDockerContextListItem } from '../../utils/dockerContextManager';
+
+export async function selectDockerContext(prompt: string): Promise<IDockerContextListItem> {
+    const fetchingContexts = localize('vscode-docker.commands.context.fetchingContexts', 'Fetching Docker contexts...');
+    let contexts: IDockerContextListItem[];
+    await window.withProgress(
```

Comment:
It should work either way but I think you can also do something like
`const contexts = await window.withProgress({ ... }, async () => dockerContextManager.listAll());`

Plus or minus an `async` or `await` somewhere in there :smile:

Reply:
Nice. Applied